examples/chart/teleport/README.md

```diff
@@ -0,0 +1,91 @@
+# Teleport
+
+[Gravitational Teleport](https://github.com/gravitational/teleport) is a modern SSH/Kubernetes API proxy server for remotely accessing clusters of Linux containers and servers via SSH, HTTPS, or Kubernetes API.
+
+## Introduction
+
+This chart deploys Teleport components to your cluster via a Kubernetes `Deployment`.
+
+By default this chart is configured as follows:
+
+- 1 replica
+- Record ssh/k8s exec and attach session to the `emptyDir` of the Teleport pod
+- The assumed externally accessible hostname of Teleport is `teleport.example.com`
+- Teleport is accessible only from within your cluster. You need `kubectl port-forward` for external access.
+- TLS is disabled on the Proxy
+
+See the comments in the default `values.yaml` and also the Teleport documentation for more options.
+
+## Prerequisites
+
+- Kubernetes 1.10+
+- A Teleport license file stored as a Kubernetes Secret object(See below)
+
+### Prepare the license file
+
+Download the `license.pem` from the Teleport dashboard, and then rename it to the filename that this chart expects:
+
+```
+cp ~/Downloads/license.pem license-enterprise.pem
+```
+
+Store it as a Kubernetes secret:
+
+```console
+kubectl create secret generic license --from-file=license-enterprise.pem
+```
+
+## Installing the chart
+
+To install the chart with the release name `teleport`, run:
+
+```
+$ helm install --name teleport ./
+```
+
+To enable TLS on the proxy, prepare the following files:
+
+- Your CA cert named `ca.pem`
+- Your proxy server cert named `proxy-server.pem`
+- Your proxy server key named `proxy-server-key.pem`
+
+Then run:
+
+```
+$ kubectl create secret tls tls-web --cert=proxy-server.pem --key=proxy-server-key.pem
+$ kubectl create configmap ca-certs --from-file=ca.pem
+$ helm install --name teleport \
+  --set proxy.tls.enabled=true \
```

Comment:
let's make it reverse to match teleport's behavior:

TLS should be always on by default, to turn on TLS, users must set `--set proxy.tls.insecure-no-tls` flag

Reply:
Thanks. I've addressed it in 9426272. If your whole purpose is just to enable TLS by default, it is a matter of setting `enabled: true` in the default `values.yaml`, hence the changes in the commit.

Regarding the configuration key, as myself being a normal helm user, I'd expect every feature toggle to be named `foo.enabled=[true|false]` rather than `proxy.tls.insecure-no-tls`. You may already know but let me show [how those toggles are named in the official helm charts](https://github.com/kubernetes/charts/search?utf8=%E2%9C%93&q=%22.enabled%22&type=).

However, you can still consider using `--set extraArgs[0]=--insecure-no-tls` you if prefer the config key to match the flag name!

Also note that an idiomatic helm chart should just work after installed without any additional values. On the other hand the teleport chart after my last commit does not. That's because you need to create tls assets in a secret and a configmap beforehand. You'll eventually want to make it work without the prerequisites. Perhaps generating tls assets automatically within the chart? Just my two cents.

Reply:
tls will get generated by default if you don't set the values in config file, so by default we should omit tls_ keys in the config file (so they will be auto generated) and turn on tls by default